audit/projects/k8s-artifacts-prod/service-accounts/k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com/iam.json

```diff
@@ -2,6 +2,7 @@
   "bindings": [
     {
       "members": [
+        "serviceAccount:k8s-infra-prow-build-trusted.svc.id.goog[test-pods/k8s-infra-gcr-promoter]",
```

Comment:
Why are we doing this?

Reply:
workload identity... 
- the pod that runs the container image promoter should run as this service account so it has permissions to promote
- the pod is scheduled via a prowjob
- the prowjob runs in k8s-infra-prow-build-trusted as of https://github.com/kubernetes/test-infra/pull/19810

(we can remove the k8s-prow IAM as a followup, will open issue)